include/proxy/http2/Http2Stream.h

```diff
@@ -80,6 +80,7 @@ class Http2Stream : public ProxyTransaction
   void set_expect_send_trailer() override;
   bool expect_receive_trailer() const override;
   void set_expect_receive_trailer() override;
+  void set_close_connection(HTTPHdr &hdr) const override;
```

Comment:
Unused

Reply:
Ah, nvm. I just realized it's override.

Reply:
Does this relay Connection header value between a request and a response? I think that should not happen.

```
Client <-- (H2) -- ATS <-- (H1) -- Origin
```

When an origin server returns `Connection: close`, ATS should not close the connection between the client and ATS, right? It's just the origin wants to close the connection between ATS and the origin. The client should be allowed to make another request, which can be served from the cache, on the same connection.

Reply:
This sets the `Connection: close`, but it doesn't decide whether the connection will be closed. `HttpTransact::handle_response_keep_alive_headers` actually decides whether that close will remain in the response or be removed. But before we never set it in the first place, so all the logic there does nothing for h2 connections.

Reply:
Ok, I think I was confused by this line. It seemed like to modify a client response based on a server response, but it doesn't.
https://github.com/apache/trafficserver/blob/cbe91698e2054ddb0ac33d34a92e270fa9194909/src/proxy/http/HttpTransact.cc#L6978

Http1Transaction and Http2Stream do the same thing in set_close_connection. It's not conditional at all. I assume Http3Transaction should do the same. Now I wonder why we don't modify the `heads` directly in HttpTransact. What's the reason for letting transactions to set the header? I think we should remove this confusing function.

Reply:
This was added in #8178 , maybe at the time we had some issues with our H2 streams and this header was causing trouble?

Reply:
So we are going back to the original behavior (do the same thing for H1 and H2). Are you sure you don't have the issue that was resolved by #8178, with this change?

If we don't have to to differentiate the behavior, I prefer the original code that modifies `heads` directly in HttpTransact.

Reply:
The intent with #8178 was to not drain the session if the HttpSM logic sets the connection close due to detected transaction failures.  Shutting down the connection in HTTP/1 makes sense since the TCP connection may be polluted in the case of bad request bodies. However, this is not an issue with HTTP/2. An error with a request body should not require shutting down the session due to cross-contamination.

Explicitly setting the Connection: close header in the HTTP/2 processing logic when w want to close the session seems reasonable if it makes our client's lives easier. But blindly interpreting Connection: Close passed from the state machine processing will cause HTTP/2 sessions to shut down needlessly in some cases.